tests/api/linearalgebra/linearalgebra.cpp

```diff
@@ -1123,70 +1119,80 @@ void test_multiply_matrix_matrix_row_major(LinearAlgebra &linear_algebra, int le
+    test_multiply_matrix_matrix_row_major(linear_algebra, large_width, large_height, unit1_width, PI, unit1, test);
+    test_multiply_matrix_matrix_row_major(linear_algebra, large_width, large_height, large_width, PI, unit1, test);
 
     // // one or more dimensions are are fraction of the encoding unit (padding required)
     int half_width = unit1_width / 2;
     int half_height = unit1_height / 2;
-    test_multiply_matrix_matrix_row_major(linear_algebra, half_width, unit1_height, unit1_width, PI, unit1);
-    test_multiply_matrix_matrix_row_major(linear_algebra, unit1_width, half_height, unit1_width, PI, unit1);
-    test_multiply_matrix_matrix_row_major(linear_algebra, unit1_width, unit1_height, half_width, PI, unit1);
-    test_multiply_matrix_matrix_row_major(linear_algebra, half_width, unit1_height, half_width, PI, unit1);
-    test_multiply_matrix_matrix_row_major(linear_algebra, unit1_width, half_height, half_width, PI, unit1);
-    test_multiply_matrix_matrix_row_major(linear_algebra, half_width, half_height, unit1_width, PI, unit1);
-    test_multiply_matrix_matrix_row_major(linear_algebra, half_width, half_height, half_width, PI, unit1);
+    test_multiply_matrix_matrix_row_major(linear_algebra, half_width, unit1_height, unit1_width, PI, unit1, test);
+    test_multiply_matrix_matrix_row_major(linear_algebra, unit1_width, half_height, unit1_width, PI, unit1, test);
+    test_multiply_matrix_matrix_row_major(linear_algebra, unit1_width, unit1_height, half_width, PI, unit1, test);
+    test_multiply_matrix_matrix_row_major(linear_algebra, half_width, unit1_height, half_width, PI, unit1, test);
+    test_multiply_matrix_matrix_row_major(linear_algebra, unit1_width, half_height, half_width, PI, unit1, test);
+    test_multiply_matrix_matrix_row_major(linear_algebra, half_width, half_height, unit1_width, PI, unit1, test);
+    test_multiply_matrix_matrix_row_major(linear_algebra, half_width, half_height, half_width, PI, unit1, test);
 
     // // some random dimensions
-    test_multiply_matrix_matrix_row_major(linear_algebra, 13, 78, 141, PI, unit1);
-    test_multiply_matrix_matrix_row_major(linear_algebra, 67, 17, 312, PI, unit1);
-    test_multiply_matrix_matrix_row_major(linear_algebra, 134, 134, 134, PI, unit1);
-    test_multiply_matrix_matrix_row_major(linear_algebra, 300, 27, 29, PI, unit1);
+    test_multiply_matrix_matrix_row_major(linear_algebra, 13, 78, 141, PI, unit1, test);
+    test_multiply_matrix_matrix_row_major(linear_algebra, 67, 17, 312, PI, unit1, test);
+    test_multiply_matrix_matrix_row_major(linear_algebra, 134, 134, 134, PI, unit1, test);
+    test_multiply_matrix_matrix_row_major(linear_algebra, 300, 27, 29, PI, unit1, test);
+}
+
+TEST(LinearAlgebraTest, MultiplyMatrixMatrix_Row_Major) {
+    RotationSet rot_instance = RotationSet(8192);
+    LinearAlgebra linear_algebra_rot = LinearAlgebra(rot_instance);
+    test_multiply_matrix_matrix_row_major_inputs(linear_algebra_rot, false);
```

Comment:
Why are we passing `false` to skip the assertions? Is this because we are not actually doing any of the math and just tracking the steps (such as rotations) used so that we can prepare for them later and then include the assertions when the actual math is done?

Reply:
Right, see above comment.